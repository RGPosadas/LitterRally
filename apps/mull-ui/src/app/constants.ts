export const ROUTES = {
  HOME: '/home',
  MAP: '/map',
  NEW_EVENT: '/new-event',
  TOOLS: '/tools',
  PROFILE: '/profile',
};
