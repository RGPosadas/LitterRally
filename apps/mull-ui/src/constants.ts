import { EventRestriction, IEvent } from '@mull/types';

/**
 * enum of the current routes for the application
 */
export enum ROUTES {
  LOGIN = '/login',
  HOME = '/home',
  MAP = '/map',
  CREATE_EVENT = '/create-event',
  TOOLS = '/tools',
  MESSAGES = '/messages',
  PROFILE = '/profile',
}

export const DAY_IN_MILLISECONDS = 86400000;

export const dummyEvent: IEvent = {
  id: 1,
  title: 'Test title',
  description:
    'Lorem ipsum dolor sit amet, regione invidunt democritum vim in, movet antiopam gubergren ne per. Est nibh magna scribentur ea, vel te munere deseruisse disputationi, eros meis ludus ne sea. Odio prompta legendos in sea, ut mei scripta labores theophrastus, id molestie probatus periculis mea. Noluisse invenire splendide sed ne, at erat quando laudem nec, possim apeirian vix at. At eam animal efficiendi interpretaris, eirmod offendit adversarium per et, summo qualisque efficiendi in has. Vel detraxit accusata ea.',
  startDate: new Date(2000, 0, 1, 0, 0, 0, 0),
  endDate: new Date(2000, 0, 2, 0, 0, 0, 0),
  location: {
    id: 1,
    point: '1260 Remembrance Road Montreal, Qc',
  },
  restriction: EventRestriction.NONE,
  participants: [
    {
      name: 'User 1',
    },
    {
      name: 'User 2',
    },
  ],
  host: {
    name: 'Test user',
  },
};
