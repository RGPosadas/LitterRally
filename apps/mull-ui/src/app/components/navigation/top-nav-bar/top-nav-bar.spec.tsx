import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ROUTES } from '../../../../constants';
import TopNavBar from './top-nav-bar';

describe('TopNavBar', () => {
  it('should render successfully', () => {
    const history = createMemoryHistory();

    const { baseElement } = render(
      <Router history={history}>
        <TopNavBar />
      </Router>
    );

    expect(baseElement).toBeTruthy();
  });

  it('button associated with current route should be active ', () => {
    const testIds = {
      [ROUTES.HOME]: 'home-navlink',
      [ROUTES.MAP]: 'map-navlink',
      [ROUTES.CREATE_EVENT]: 'create-event-navlink',
      [ROUTES.CAMERA]: 'camera-navlink',
      [ROUTES.PROFILE.DISPLAY]: 'profile-desktop-navlink',
      [ROUTES.MESSAGES]: 'messages-navlink',
    };

    const history = createMemoryHistory();
    const dom = render(
      <Router history={history}>
        <TopNavBar />
      </Router>
    );

    for (const [key, value] of Object.entries(testIds)) {
      history.push(key);
      const element = dom.getByTestId(value);
      expect(element.classList.contains('active')).toBeTruthy();
    }
  });

  it('should match snapshot', () => {
    const history = createMemoryHistory();

    const tree = renderer
      .create(
        <Router history={history}>
          <TopNavBar />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
