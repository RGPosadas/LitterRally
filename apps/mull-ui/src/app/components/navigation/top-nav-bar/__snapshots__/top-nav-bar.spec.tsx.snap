// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNavBar should match snapshot 1`] = `
<div
  className="top-nav-container top-nav-bar-shadow"
>
  <img
    alt="Mull logo"
    className="logo nav-element"
    src="mull-logo.png"
  />
  <div
    className="extra-buttons"
  >
    <div
      className="nav-buttons-container"
    >
      <a
        aria-current={null}
        data-testid="home-navlink"
        href="/home"
        onClick={[Function]}
      >
        <svg
          className="nav-button"
        >
          HomeIcon.svg
        </svg>
      </a>
      <a
        aria-current={null}
        data-testid="map-navlink"
        href="/map"
        onClick={[Function]}
      >
        <svg
          className="nav-button"
        >
          MapIcon.svg
        </svg>
      </a>
      <a
        aria-current={null}
        data-testid="create-event-navlink"
        href="/create-event"
        onClick={[Function]}
      >
        <svg
          className="nav-middle-button"
        >
          CreateEventIcon.svg
        </svg>
      </a>
      <a
        aria-current={null}
        data-testid="tools-navlink"
        href="/tools"
        onClick={[Function]}
      >
        <svg
          className="nav-button"
        >
          MachineLearningIcon.svg
        </svg>
      </a>
      <a
        aria-current={null}
        data-testid="messages-navlink"
        href="/messages"
        onClick={[Function]}
      >
        <svg
          className="nav-button"
        >
          MessagesIcon.svg
        </svg>
      </a>
    </div>
  </div>
  <a
    aria-current={null}
    className="profile-button"
    data-testid="profile-desktop-navlink"
    href="/profile"
    onClick={[Function]}
  >
    <svg
      className="nav-button"
    >
      ProfileIcon.svg
    </svg>
  </a>
</div>
`;
